src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/Vector128_1.cs

```diff
@@ -13,13 +16,25 @@ namespace System.Runtime.Intrinsics
     [DebuggerDisplay("{DisplayString,nq}")]
     [DebuggerTypeProxy(typeof(Vector128DebugView<>))]
     [StructLayout(LayoutKind.Sequential, Size = Vector128.Size)]
-    public readonly struct Vector128<T> where T : struct
+    public readonly struct Vector128<T> : IEquatable<Vector128<T>>, IFormattable
+        where T : struct
     {
         // These fields exist to ensure the alignment is 8, rather than 1.
         // This also allows the debug view to work https://github.com/dotnet/coreclr/issues/15694)
         private readonly ulong _00;
         private readonly ulong _01;
 
+        /// <summary>Gets the number of <typeparamref name="T" /> that are in a <see cref="Vector128{T}" />.</summary>
+        /// <exception cref="NotSupportedException">The type of the current instance (<typeparamref name="T" />) is not supported.</exception>
+        public static int Count
+        {
+            get
+            {
+                ThrowIfUnsupportedType();
+                return Vector128.Size / Unsafe.SizeOf<T>();
```

Comment:
`Unsafe.SizeOf` seems not able to be folded by Roslyn, so marking the getter with `[intrinsic]` and expanding it in JIT can enable more optimizations.

Reply:
I want to get the CoreFX side changes up first so I can get test in before enabling intrinsic support.

Reply:
What is the current implementation? Is it not a pattern that could be recognized to avoid adding yet more intrinsics?

Reply:
The `Unsafe.SizeOf` used by `S.P.Corelib` is an `Intrinsic` already: https://github.com/dotnet/coreclr/blob/4a6753dcacf44df6a8e91b91029e4b7a4f12d917/src/System.Private.CoreLib/shared/Internal/Runtime/CompilerServices/Unsafe.cs#L57

The `Unsafe.SizeOf` that ships publicly just executes `sizeof !!T`: https://github.com/dotnet/corefx/blob/master/src/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.il#L154

In both cases, I would hope that the JIT recognizes this as a constant and does the appropriate folding.
 -- I may want to check if `AggressiveInlining` is needed, however. Although the JIT has generally done a good job for cases as small as this (in my experience)